### PR TITLE
[CLOUDTRUST-2956] Add KYC endpoint to send an enrolment SMS code

### PR DESCRIPTION
**Pull request by fperot74**

### CODE REVIEW

**Review by fratt-elca — APPROVED** [review]
Retest ok.
Please clarify with RPO if we need the Corporate KYC endpoint, I didn't use it in the front-end.

**Comment by fratt-elca** [comment]
Merge at the same time as https://bitbucket.svc.elca.ch/projects/CLOUDTRUST/repos/2019.oc.backoffice/pull-requests/353

File: pkg/kyc/authorization.go [hunk 1]
```diff
@@ -29,6 +29,7 @@ var (
 	KYCGetUserByUsernameInSocialRealm = newAction("KYC_GetUserByUsernameInSocialRealm", security.ScopeRealm)
 	KYCValidateUserInSocialRealm      = newAction("KYC_ValidateUserInSocialRealm", security.ScopeRealm)
 	KYCValidateUser                   = newAction("KYC_ValidateUser", security.ScopeGroup)
+	KYCSendSmsCodeInSocialRealm       = newAction("KYC_SendSmsCodeInSocialRealm", security.ScopeGroup)
```
Comment (harture) [inline pkg/kyc/authorization.go:32]: ScopeRealm, as we target social realm we cannot check the group of the target

File: pkg/kyc/component.go [hunk 2]
```diff
@@ -326,3 +328,22 @@ func (c *component) reportEvent(ctx context.Context, apiCall string, values ...s
...
 }
+
+func (c *component) SendSmsCodeInSocialRealm(ctx context.Context, userID string) (string, error) {
+	return c.sendSmsCode(ctx, c.socialRealmName, userID)
+}
+
+func (c *component) sendSmsCode(ctx context.Context, realm string, userID string) (string, error) {
+	var accessToken = ctx.Value(cs.CtContextAccessToken).(string)
+
+	smsCodeKc, err := c.keycloakClient.SendSmsCode(accessToken, realm, userID)
```
Comment (harture) [inline pkg/kyc/component.go:339]: our keycloak API do not restrict the realm used to send the SMS ?
I would have expected that we need to use technical user to be able to send SMS code in a cross realm.

**Pull request merged**